### PR TITLE
Validate audit policy api version on shoot update request

### LINKED ISSUE #6882: After Kubernetes update API server cannot start during rolling update because of outdated audit policy api version.
**How to categorize this issue?**

/area control-plane
/kind impediment

**What happened**:

Cluster stuck in deletion workflow, because API server deployment was not healthy.

The API server deployment was rolled according to a pending kubernetes upgrade.
The new replicaset couldn't succeed, because a customer configured audit policy
used an outdated version (v1beta). Therefore the deployment never gets ready, and the deletion is stuck.

The config map providing the policy cannot be changed directly in the cluster, therefore
it must be fixed in the project namespace in the garden cluster.

Fortunately v1 and v1beta are compatible, so the problem resolved after just changing the api version
in the original config map.

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:

- Gardener version: canary

### PR DESCRIPTION
**Pull request by acumino**
**How to categorize this PR?**

/area control-plane
/kind enhancement

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes https://github.com/gardener/gardener/issues/6882

**Special notes for your reviewer**:

**Release note**:

```feature user
`gardener-admission-controller` now validates `Shoot` Kubernetes version compatibility with Audit Policy API version on `Shoot` update request.
```

### CODE REVIEW

**Comment by rfranzke** [comment]
> /cc @rfranzke is cherry-picking this PR required?

No strong opinion, technically yes since it fixes a bug, but OTOH it was not working for a long time and there is no "pressure".

**Comment by acumino** [comment]
> > /cc @rfranzke is cherry-picking this PR required?
> 
> No strong opinion, technically yes since it fixes a bug, but OTOH it was not working for a long time and there is no "pressure".

Ok, then there is not need to merge this PR in this release.

File: pkg/admissioncontroller/webhook/admission/auditpolicy/handler_test.go [hunk 1]
```diff
@@ -540,6 +541,22 @@ rules:
 				test(admissionv1.Create, nil, shootv1beta1K8sV124, false, statusCodeInvalid, "audit policy with apiVersion 'v1alpha1' is not supported for kubernetes version >= 1.24.0", "")
 			})
 
+			It("references a valid auditPolicy/v1alhpa1 (UPDATE kubernetes version)", func() {
```
Comment (shafeeqes) [inline pkg/admissioncontroller/webhook/admission/auditpolicy/handler_test.go]:
```suggestion
			It("references a valid auditPolicy/v1alpha1 (UPDATE kubernetes version)", func() {
```
and other occurrences as well.

File: pkg/admissioncontroller/webhook/admission/auditpolicy/handler_test.go [hunk 2]
```diff
@@ -540,6 +541,22 @@ rules:
 				test(admissionv1.Create, nil, shootv1beta1K8sV124, false, statusCodeInvalid, "audit policy with apiVersion 'v1alpha1' is not supported for kubernetes version >= 1.24.0", "")
 			})
 
+			It("references a valid auditPolicy/v1alpha1 (UPDATE kubernetes version)", func() {
+				returnedCm := v1.ConfigMap{
```
Comment (rfranzke) [inline pkg/admissioncontroller/webhook/admission/auditpolicy/handler_test.go]:
```suggestion
				returnedCm := corev1.ConfigMap{
```

**Review by rfranzke — COMMENTED** [review]
Thanks, looks good.
One thing (unrelated to this PR, but since you are already on it...)

**Review by rfranzke — COMMENTED** [review]
/lgtm

**Comment by oliver-goetz** [comment]
It's a fix and it is quite small, so let's include it into v1.62 👍 
/milestone v1.62

**Pull request merged**